RickAndMortyFunApp/CharacterViewController.swift

```diff
@@ -12,10 +12,16 @@ class CharacterViewController: UIViewController {
 
     private let cellId = "CellId"
     var tableView = UITableView()
+    var characters: [Character]?
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setupTableView()
+        let network = NetworkService()
```

Comment:
This is not good. We want to keep reference in class level, because we can lose reference before we get response. Also try to inject it and not to create instance `var nerworkService: NetworkService!`